pkg/health/health.go

```diff
@@ -0,0 +1,46 @@
+package health
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spiffe-helper/pkg/sidecar"
+)
+
+type CheckConfig struct {
+	ListenerEnabled bool   `hcl:"listener_enabled"`
+	BindPort        int    `hcl:"bind_port"`
+	HealthPath      string `hcl:"health_path"`
+}
+
+func StartHealthServer(daemonMode bool, healthCheckConfig CheckConfig, log logrus.FieldLogger, sidecar *sidecar.Sidecar) error {
+	if daemonMode && healthCheckConfig.ListenerEnabled {
+		http.HandleFunc(healthCheckConfig.HealthPath, func(w http.ResponseWriter, _ *http.Request) {
+			healthy := sidecar.CheckHealth()
+			if healthy {
+				_, err := w.Write([]byte(http.StatusText(http.StatusOK)))
+				log.Error(err)
+				if err != nil {
+					return
+				}
```

Comment:
- Should this return err if not nil?
- Is the log.Error(err) intended or a debugging thing?

Reply:
oh nm, its an embedded function. In the case of an error should it return a status code?

Reply:
I'm not sure how to do that without a `Write()` which is what failed in this case. I'm also not sure when that could happen.